File: lookbehind/lookBehind.js
"use strict";

function solution(needle, haystack, greedy = true) {
  let size = needle.length;
  let start = size - 1;
  let token = haystack.slice(0, size);
  let i;
  let found = [];

  for (i = start; i < haystack.length; i++) {
    if (token === needle) {
      found.push(i - start);
      if (!greedy) {
        break;
      }
    }

    if (haystack[i + 1]) {
      token = token.slice(1) + haystack[i + 1];
    }
  }

  return found;
}

// var txt = "ABABDABACADABABCABAB"; //11
// var pat = "ABABCABAB";

var txt = "AABAACAADAABAABA"; //0,9,12
var pat = "AABA";
console.log(solution(pat, txt, true));

File: lookbehind/lookBehind.py
def lookbehind(needle, heystack, greedy=True):
    size = len(needle)
    start = size - 1
    token= heystack[0:size]
    found = []

    for i in range(start,len(heystack)):
        if token == needle:
            found.append(i-start)
            if greedy == False:
                break
        if  i < len(heystack)-1:   
            token = token[1:] + heystack[i+1]


    return found

txt = "AABAACAADAABAABA"; #11
pat = "AABA";

print(lookbehind(pat,txt, True))